lib/dry/system/container.rb

```diff
@@ -249,7 +249,7 @@ def finalize!(&block)
           manual_registrar.finalize!
           auto_registrar.finalize!
 
-          freeze
+          self.freeze if freeze
```

Comment:
this will return `nil` if it's frozen already, so it needs to be `freeze ? self.freeze : self`